src/CMakeLists.txt

```diff
@@ -141,3 +141,21 @@ target_link_libraries(${project_name} ${global_libraries})
 install(TARGETS ${project_name}
   RUNTIME DESTINATION bin
 )
+
+find_package(Plantuml)
+if(PLANTUML_FOUND)
+#message("plantuml found PLANTUML_JARFILE=${PLANTUML_JARFILE}")
+else(Plantuml_FOUND)
+message("plantuml not found. No UML diagrams are generated. You may define the environment variable PLANTUML_PATH...")
```

Comment:
Especially behaviour diagrams may help to provide graphical information how different components work together (e.g. using activity diagrams). See: http://goto40.github.io/create_anaglyph/page_architecture.html ("Runtime View")